### PR TITLE
trottersuzuki recipe added

### PR DESCRIPTION
**Pull request by peterwittek**

### CODE REVIEW

File: recipes/trottersuzuki/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+
+test:
+  imports:
+    - trottersuzuki
+
+about:
+  home: https://github.com/trotter-suzuki-mpi/trotter-suzuki-mpi
+  license: GPL-3.0
+  license_family: GPL
+  license_file: LICENSE
```
Comment (dougalsutherland) [inline recipes/trottersuzuki/meta.yaml]: The license file in the distribution is named `COPYING`, not `LICENSE`.

File: recipes/trottersuzuki/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+  number: 0
+  skip: True  # [win]
+  script: python setup.py install --single-version-externally-managed --record record.txt
+
+requirements:
+  build:
+    - python
+    - setuptools
+    - gcc
+    - numpy
```
Comment (dougalsutherland) [inline recipes/trottersuzuki/meta.yaml]: This package links to numpy, so you'll need to version this [as listed in the docs](https://conda-forge.org/docs/meta.html#building-against-numpy).

File: recipes/trottersuzuki/meta.yaml (hunk 1 above)
Comment (peterwittek) [inline recipes/trottersuzuki/meta.yaml]: Well spotted, thanks.

File: recipes/trottersuzuki/meta.yaml (hunk 2 above)
Comment (peterwittek) [inline recipes/trottersuzuki/meta.yaml]: Fixed, thanks.

**Comment by dougalsutherland** [comment]
The Travis build is failing because if you're building with gcc, you need to have `libgcc` at runtime too. You might look into whether the recipe would work with [the openmp package](https://github.com/conda-forge/openmp-feedstock) instead....

**Comment by peterwittek** [comment]
Adding `libgcc` to runtime resolved this, but while we are at it, let me bounce two questions on you (both are relevant to PR #3383 as well):

1. Do you have an example of using the OpenMP package on OS X instead of gcc?

2. Building with CUDA is a bit tricky. The script `setup.py` looks for the `CUDAHOME` environment variable, and if it is set, it uses `nvcc` from there. Conda-build is smart enough that this does cause any problems even if the system GCC has priority in the path, so this solution is already fairly robust. On the other hand, I wonder if there is an easier way to do this from the user's perspective.

Many thanks for the comments so far.

**Comment by dougalsutherland** [comment]
1. It looks like [xgboost](https://github.com/conda-forge/xgboost-feedstock/tree/master/recipe) and [lightgbm](https://github.com/conda-forge/lightgbm-feedstock/tree/master/recipe) use it. I've been meaning to look into it (https://github.com/conda-forge/shogun-cpp-feedstock/issues/2) and haven't yet, but at a glance it looks like just making sure it compiles with clang and maybe adding `$PREFIX/lib` to `LIBRARY_PATH` or similar.

2. I'm not sure whether compiling with cuda on conda-forge is possible; the main discussion is at https://github.com/conda-forge/conda-forge.github.io/issues/63.

**Comment by peterwittek** [comment]
This is really cool, I had no idea Clang came such a long way. I also spotted that now it supports CUDA. These changes will require a revision of the build scripts, so I leave it there for now, but I opened two respective issues. I patched the OS X version to use OpenMP with the conda GCC. I am happy with the recipe now.

File: recipes/trottersuzuki/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+    - gcc
+    - numpy >=1.10
+  run:
+    - python
+    - libgcc
+    - numpy >=1.10
+
+test:
+  imports:
+    - trottersuzuki
```
Comment (ocefpaf) [inline recipes/trottersuzuki/meta.yaml]: Please add the `conda-inspect` test like https://github.com/conda-forge/nco-feedstock/blob/master/recipe/meta.yaml#L47-L48

It helps us to investigate the linkages.

File: recipes/trottersuzuki/meta.yaml [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+  number: 0
+  skip: True  # [win]
+  script: python setup.py install --single-version-externally-managed --record record.txt
+
+requirements:
+  build:
+    - python
+    - setuptools
+    - gcc
+    - numpy >=1.10
```
Comment (ocefpaf) [inline recipes/trottersuzuki/meta.yaml]: If you want to support all numpy's from 1.10 this should be `numpy 1.10.*` (The run section is fine with the `>=`.)

**Comment by dougalsutherland** [comment]
I think you also need to include the `openmp`, `clangdev`, and `libcxx` packages at build time on osx, and `openmp` and `libcxx` at runtime: see [how xgboost does it](https://github.com/conda-forge/xgboost-feedstock/blob/cbd3078becfe5bf8f55c35144cfd0080353cca96/recipe/meta.yaml#L27-L36). The Travis build environment won't have these by default, and even if it did users will need the libraries at runtime.

Question for @conda-forge/openmp though: does the `openmp` package conflict with `libgcc`, since both include a `lib/libgomp.dylib` file? How does that work out in practice?

**Comment by peterwittek** [comment]
At the moment, I would like to continue using GCC for compiling with OpenMP on OS X. I am glad that Clang finally has decent OpenMP support, but we would need further testing for a decent support (trotter-suzuki-mpi/trotter-suzuki-mpi#163). Travis builds it and tests it without problems with OpenMP enabled using GCC.

**Pull request merged**